File: Interpolacja Lagrange'a.py
def interpolation(listX, listY, x, n):
    list_pom = []

    for k in range(n):
        list_pom.append(1)

    for i in range(n):
        for j in range(n):
            if i != j:
                list_pom[i] *= (x - listX[j])/(listX[i] - listX[j])

    print(list_pom)

    val = 0
    for a in range(n):
        val += listY[a] * list_pom[a]

    print("Szukana wartosc funkcji w argumencie x = ", x, "jest rowna y = ", val)


if __name__ == '__main__':
    numberOfPoints = int(input("Podaj liczbe argumentow dla ktorych znasz wartosci: "))
    valueInCertainArgument = int(input("Podaj argument dla ktorego chcesz znalezc wartosc: "))

    list_of_args = []
    list_of_vals = []

    for i in range(numberOfPoints):
        x = int(input("Podaj argument dla " + str(i + 1) + " punktu: "))
        y = int(input("Podaj wartosc dla " + str(i + 1) + " punktu: "))
        list_of_args.append(x)
        list_of_vals.append(y)
        #list_of_points.append((x,y))

    """print("X: ")
    print(list_of_args)
    print("Y: ")
    print(list_of_vals)"""

    interpolation(list_of_args, list_of_vals, valueInCertainArgument, numberOfPoints)

File: Interpolacja Lagrange'a.cpp

#include <iostream>

using namespace std;

void Lab2(int n, double* x, double* y, double szukana) {
    double* l = new double[n], result = 0;

    for (int q = 0; q < n; q++) {
        l[q] = 1;
    }

    for (int q = 0; q < n; q++)
        for (int w = 0; w < n; w++)
            if (q != w)
                l[q] *= (szukana - x[w]) / (x[q] - x[w]);

    for (int q = 0; q < n; q++) {
        cout << l[q] << endl;
    }


    for (int q = 0; q < n; q++) {
        result += y[q] * l[q];
    }

    cout << "Wartosc funkcji w punkcie x=" << szukana << " jest rowna y="
         << result << "\n";
}

int main()
{
    cout << "Podaj liczbe punktow, dla ktorych sa znane wartosci oraz wartosc dla ktorej chcesz szukac: \n";
    int n, szukana;
    cin >> n >> szukana;
    double* x = new double[n];
    double* y = new double[n];

    for (int q = 0; q < n; q++) {
        cout << "Podaj kolejno punkt i jego wartosc dla wezla nr." << q + 1 << "\n";
        cin >> x[q] >> y[q];
    }

    Lab2(n, x, y, szukana);
    system("pause");
    return 0;
}